website/app/core/container.server.ts

```diff
@@ -0,0 +1,125 @@
+import { createContainer, InjectionMode, asFunction, asValue } from 'awilix';
+import type { Logger } from '~/domain/contracts/logger';
+import { createLogger } from '~/infrastructure/create-logger.server';
+import { createCommandBus, createQueryBus } from 'missive.js';
+import { CommandBus, CommandDefitions, QueryBus, QueryDefitions } from '~/domain/contracts/bus';
+import { MainConfig, MainConfigSchema } from '~/domain/contracts/config';
+import { validateConfigOrExit } from './configs.server';
+import crypto from 'node:crypto';
+import { createSessionManager } from '~/infrastructure/session/create-session-manager.server';
+import { redirect } from '@remix-run/node';
+import { SessionManager } from '~/domain/contracts/session';
+import { createMailer } from '~/infrastructure/create-mailer.server';
+import { Mailer } from '~/domain/contracts/mailer';
+import { createInngestServer } from '~/infrastructure/inngest/create-inngest.server';
+import { InngestServer } from '~/domain/contracts/inngest';
+import { buildInngestFunctions } from '~/infrastructure/inngest/functions.server';
+import { createServerTimingsProfiler } from '~/infrastructure/create-server-timings-profiler.server';
+import { PocketBaseClient } from '~/domain/contracts/pocketbase';
+import { createPocketbaseClient } from '~/infrastructure/pocketbase/create-pocketbase-client.server';
+import { createFetchSpeakersWithTalksHandler } from '~/domain/use-cases/fetch-speakers-with-talks.server';
+import { createLoadEventDetailsHandler } from '~/domain/use-cases/load-event-details.server';
+import { createLumaClient } from '~/infrastructure/luma/create-luma-client.server';
+import { createPosthogClient } from '~/infrastructure/posthog/create-posthog-client.server';
+
+export const buildContainer = () => {
+  const port = process.env.PORT ? Number.parseInt(process.env.PORT, 10) : 3000;
+
+  // 1. load the config
+  const config: MainConfig = validateConfigOrExit(MainConfigSchema, {
+    instanceId: process.env.INSTANCE_ID || crypto.randomUUID(),
+    logLevel: ['info', 'debug'],
+    port,
+    environment: process.env.NODE_ENV || 'development',
+    origin: process.env.ORIGIN || `http://localhost:${port}`,
+    pocketbase: {
+      origin: process.env.POCKETBASE_ORIGIN || '',
+      publicOrigin: process.env.PUBLIC_POCKETBASE_ORIGIN || '',
+      adminEmail: process.env.POCKETBASE_EMAIL || '',
+      adminPassword: process.env.POCKETBASE_PASSWORD || '',
+    },
+    inngest: {
+      signingKey: process.env.INNGEST_SIGNING_KEY,
+      eventKey: process.env.INNGEST_EVENT_KEY,
+    },
+    sessionSecret: process.env.SESSION_SECRET || '',
```

Comment:
it's WIP here, we should not have so many `|| ''` and `|| 'xxx'` should be optional when it's optional or required but most likely not default value unless it makes sense.

Reply:
Can we fix this before merging? Remove the `|| ''` and `|| 'xxx'` everywhere and make sure that optional configuration options properly disable the services (like resend) when not set instead of throwing.